src/cockpit/internal_endpoints.py

```diff
@@ -145,9 +146,17 @@ def __init__(self):
         self.name = user.pw_name
         self.full = user.pw_gecos
         self.id = user.pw_uid
+        self.gid = user.pw_gid
         self.home = user.pw_dir
         self.shell = user.pw_shell
-        self.groups = [gr.gr_name for gr in grp.getgrall() if user.pw_name in gr.gr_mem]
+
+        # We want the primary group first in the list, without duplicates.
+        # This is a bit awkward because `set()` is unordered...
+        groups = [grp.getgrgid(user.pw_gid).gr_name]
```

Comment:
afaik, this can throw, when you have no group name set.

Reply:
This will throw if the user's gid is not present in the groups file.  We can also throw a few lines up, though, if their uid is not present in the passwd file, so...

Reply:
EH, I was wrong when the group name is non-existant this just returns `''`. So imo all not a concern.